ui/src/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -169,7 +170,50 @@ export function WorkflowLogsViewer({workflow, initialNodeId, initialPodName, con
     // default to the node id of the pod
     const nodeId = initialNodeId || podNamesToNodeIDs.get(podName);
     const node = workflow.status.nodes[nodeId];
-    const templates = execSpec(workflow).templates.filter(t => !node || t.name === getTemplateNameFromNode(node));
+
+    useEffect(() => {
+        if (!node || !node.templateRef) {
+            setTemplateFromRef(null);
+            return;
+        }
+
+        const fetchTemplate = async () => {
+            try {
+                if (node.templateRef.clusterScope) {
+                    const tmpl = node.templateRef.clusterScope
+                        ? await services.clusterWorkflowTemplate.get(node.templateRef.name)
+                        : await services.workflowTemplate.get(node.templateRef.name, workflow.metadata.namespace);
+                    const matchingTemplate = tmpl.spec.templates.find(t => t.name === node.templateRef.template);
+
+                    if (matchingTemplate) {
+                        setTemplateFromRef(matchingTemplate);
+                    } else {
+                        setTemplateFromRef(null);
+                    }
+                } else {
+                    const tmpl = await services.workflowTemplate.get(node.templateRef.name, workflow.metadata.namespace);
+                    const matchingTemplate = tmpl.spec.templates.find(t => t.name === node.templateRef.template);
+
+                    if (matchingTemplate) {
+                        setTemplateFromRef(matchingTemplate);
+                    } else {
+                        setTemplateFromRef(null);
+                    }
+                }
```

Comment:
You could remove the if..else and do

```suggestion
                    const tmpl = node.templateRef.clusterScope
                        ? await services.clusterWorkflowTemplate.get(node.templateRef.name)
                        : await services.workflowTemplate.get(node.templateRef.name, workflow.metadata.namespace);
                    const matchingTemplate = tmpl.spec.templates.find(t => t.name === node.templateRef.template);

                    if (matchingTemplate) {
                        setTemplateFromRef(matchingTemplate);
                    } else {
                        setTemplateFromRef(null);
                    }
```